File: engine/engine-model/src/main/scala/tools/packet/serverlist/ServerIP.scala
package tools.packet.serverlist

import java.net.InetAddress

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ServerIP(private var _inetAddress: InetAddress, private var _port: Int, private var _clientId: Int) extends PacketInput {
  def inetAddress: InetAddress = _inetAddress

  def port: Int = _port

  def clientId: Int = _clientId

  override def opcode(): SendOpcode = SendOpcode.SERVER_IP
}

File: engine/engine-model/src/main/java/tools/packet/serverlist/ServerIP.java
package tools.packet.serverlist;

import java.net.InetAddress;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ServerIP(InetAddress inetAddress, Integer port, Integer clientId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SERVER_IP;
   }
}